cmd/artifact/follow/follow.go

```diff
@@ -261,7 +261,7 @@ func (o *artifactFollowOptions) RunArtifactFollow(ctx context.Context, args []st
 		} else {
 			o.Printer.Info.Printfln("Creating follower for %q, with check every %s", a, o.every.String())
 		}
-		ref, err := utils.ParseReference(mergedIndexes, a)
+		ref, err := o.IndexCache.ParseReference(a)
```

Comment:
Better than before, but it seems a bit weird to me that `ParseReference(a)` belongs to a _cache_ thing.
Not a blocker, anyway.

Reply:
The *cache* thing here is and abstraction built upon the indexes/mergeIndexes. We need the *mergedIndexes* info in order to "calculate" which reference to use for an artifact. I agree that we should change the name of the function instead of moving it around.